content/Android Arc Components - Amaury Ricard.md

```diff
@@ -0,0 +1,137 @@
+
+## Introduction to Android Architecture Components
+
+According to https://developer.android.com:
+"Android architecture components are a collection of libraries that help you design robust, testable, and maintainable apps. Start with classes for managing your UI component lifecycle and handling data persistence."
+
+Android Architecture Components are focused to solve many problems that Android developers face on a project. It focuses on a clean and easy to maintain architecture such as MVVM, and trying to take advantage of all the capabilities of the Android framework and how it is structured.
+
+![ArchComponets](images/arch_componets.png)
+
+This post is to give a brief introduction to Android Architecture Components, in later posts we will delve deeper into each of these libraries aiming to show the capabilities of each one.
+
+Let's start:
+
+- ### Life cycle:
+
+Android introduce lifecycle as a way to avoid memory leaks and keep a cleaner code, and each activity|fragment|lifecycle-owner can be abstracted from the behavior of an object at each stage of its life cycle, that object can subscribe to the activity|fragment lifecycle implementing the interface androidx.lifecycle in order to execute certain actions in each state:
+
+![LifeCycle](images/lifecycle.jpg)
+
+To subscribe to a lifecycle state you can use annotations like:
+
+``` kotlin
+  @OnLifecycleEvent(Lifecycle.Event.ON_RESUME)
+  fun yourOnResumeActionListener() {
+	...	
+  }
+```
+
+Eache states:
+ON_RESUME,ON_START,ON_STOP,ON_DESTROY,ON_PAUSE,ON_CREATE
+Note: Each activity|fragment|lifecycle-owner needs to add the object to his own lifecycle observers:
+
+``` kotlin 
+    this.getLifecycle().addObserver(objectWithLifeCycleListener)
+```
+
+- ### Live Data:
+
+This concept is not new for android, this is a new way to see and observable object that was previously introduced with "androidx". With live data they introduce it a owner in order to know when to notify data depending of the owner lifecycle in order to manage consistent data and avid memory leaks. 
```

Comment:
"introduce it TO AN owner", and also "data depending ON the owner". In the last line, it should be "avoid", it's missing an O.